Core Templates should not be assigned to infoj fields

simon-leech opened the pull request:
In the `infojMap` method we ensure that core templates cannot be assigned to the `workspace.templates` and `field` lookup. 
This prevents accidentally overwriting of a field such as a field called `cluster` with the core `cluster` query. 

## GitHub Issue

#1889 

## Type of Change
Please delete options that are not relevant, and select all options that apply. 

- ✅ Bug fix (non-breaking change which fixes an issue)

## Testing Checklist 
Please delete options that are not relevant, and select all options that apply. 

- ✅ Existing Tests still pass
- ✅ Updated Existing Tests
- ✅ Ran locally on my machine

## Code Quality Checklist
Please delete options that are not relevant, and select all options that apply. 

- ✅ My code follows the guidelines of XYZ
- ✅ My code has been commented
- ✅ Documentation has been updated
- ✅ New and existing unit tests pass locally with my changes

AlexanderGeere commented:
@simon-leech The value was not being assigned the field if it is called `cluster` i added the value assignment before the continue. i tested using `bugs_testing\ui_elements\workspace.json` I added a field called cluster to the `scratch_wkt` layer.

simon-leech commented:
> @simon-leech The value was not being assigned the field if it is called `cluster` i added the value assignment before the continue. i tested using `bugs_testing\ui_elements\workspace.json` I added a field called cluster to the `scratch_wkt` layer.

Thanks @AlexanderGeere  - good spot! This is working as expected for me and correctly returns the field in the get request

AlexanderGeere reviewed (approved):
Im happy with this, but someone else should probably take a lot
